pkg/setup/health.go

```diff
@@ -153,6 +168,7 @@ type HealthServer struct {
 	parts          map[string]report
 	mx             sync.Mutex
 	BackOffFactory func() backoff.BackOff
+	Clock          func() time.Time
 }
 
 func (h *HealthServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
i think we need to consider the ttl here as well

Reply:
Yes, we do. We can do that pretty well in the reports method.

Reply:
done